x-pack/plugins/observability_ai_assistant/public/utils/get_timeline_items_from_conversation.tsx

```diff
@@ -73,50 +95,59 @@ export function getTimelineItemsfromConversation({
           break;
 
         case MessageRole.User:
+          canCopy = true;
+          canGiveFeedback = false;
+          canRegenerate = false;
+          hide = false;
           // User executed a function:
-          if (message.message.name && functionCall) {
-            title = i18n.translate('xpack.observabilityAiAssistant.executedFunctionEvent', {
-              defaultMessage: 'executed the function {functionName}',
-              values: {
-                functionName: message.message.name,
-              },
-            });
-
-            content = convertFunctionParamsToMarkdownCodeBlock({
-              name: message.message.name,
-              arguments: JSON.parse(functionCall.arguments || '{}'),
-            });
 
-            element = chatService.hasRenderFunction(message.message.name) ? (
-              <RenderFunction
-                name={message.message.name}
-                arguments={functionCall?.arguments}
-                response={message.message}
-              />
-            ) : null;
+          if (message.message.name && functionCall) {
+            const parsedContent = JSON.parse(message.message.content ?? 'null');
+            const isError = !!(parsedContent && 'error' in parsedContent);
+
+            title = !isError
+              ? i18n.translate('xpack.observabilityAiAssistant.executedFunctionEvent', {
+                  defaultMessage: 'executed the function {functionName}',
+                  values: {
+                    functionName: message.message.name,
+                  },
+                })
+              : i18n.translate('xpack.observabilityAiAssistant.executedFunctionFailureEvent', {
```

Comment:
explicitly mention a function call has failed. maybe there should be some additional highlighting here.